### PR TITLE
Da http API

### PR DESCRIPTION
**Pull request by al8n**
Add DA API based on axum.

### CODE REVIEW

File: nomos-services/mempool/src/backend/mod.rs [hunk 1]
```diff
@@ -50,9 +50,22 @@ pub trait MemPool {
...
     Pending,
+    /// Rejected status
     Rejected,
+    /// Accepted status
+    ///
+    /// The block id of the block that contains the item
+    #[cfg_attr(
+        feature = "openapi",
+        schema(
+            example = "e.g. [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]"
```
Comment (zeegomo) [inline nomos-services/mempool/src/backend/mod.rs:67]: nit: should we use hex representation? this is a bit cumbersome to write

File: nomos-http-api/src/http/backend/axum.rs [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+#[utoipa::path(
+    get,
+    path = "/da/metrics",
+    responses(
+        (status = 200, description = "Get the mempool metrics of the da service", body = MempoolMetrics),
+        (status = 500, description = "Internal server error", body = String),
+    )
+)]
+async fn da_metrics(State(store): State<Store>) -> impl IntoResponse {
```
Comment (zeegomo) [inline nomos-http-api/src/http/backend/axum.rs:80]: we probably needs something similar for the other mempool as well

File: nomos-http-api/src/http/backend/axum.rs [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    tags(
+        (name = "da", description = "data availibility related APIs")
+    )
+)]
+struct ApiDoc;
+
+type Store = Arc<AxumBackendSettings>;
+
+#[async_trait::async_trait]
+impl Backend for AxumBackend {
```
Comment (zeegomo) [inline nomos-http-api/src/http/backend/axum.rs:45]: this can go in a separate PR

Comment (al8n) [reply nomos-http-api/src/http/backend/axum.rs:45]: The merged PR has a test using Axum, but it is not in the normal code.

Comment (danielSanchezQ) [reply nomos-http-api/src/http/backend/axum.rs:45]: Aaah, so yes, this axum implementation should go in a different pr.

**Comment by al8n** [comment]
Close. Split to sub PRs.
